### PR TITLE
Add RTrace support and Types

### PR DESCRIPTION
**Pull request by jtenner**
Include `rtrace` with the `assemblyscript` package.

image

- [x] modifies `package.json` to include files
- [x] adds `lib/rtrace/index.d.ts` types for tooling
- [x] adds code documentation (to the best of my ability)

All feedback is welcome.

### CODE REVIEW

File: lib/rtrace/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+ * The RTrace interface represents a collection of properties that help develoeprs describe the
+ * reference counts and state of the Web Assembly module. It is used by the AssemblyScript loader
```
Comment (dcodeIO) [inline lib/rtrace/index.d.ts]: Isn't really used by the loader, but more of an extra package to include as an import, independently of whether the loader is used or not :)

File: lib/rtrace/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/**
+ * The RTrace interface represents a collection of properties that help develoeprs describe the
```
Comment (dcodeIO) [inline lib/rtrace/index.d.ts]: "developers"

Comment (jtenner) [reply lib/rtrace/index.d.ts]: All set for review :)

File: lib/rtrace/index.d.ts (hunk 1 above)
Comment (jtenner) [reply lib/rtrace/index.d.ts]: The line was simply removed. All set for review.

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**